### PR TITLE
feature: tags field

### PR DESCRIPTION
**Pull request by adrianthedev**
# Description

Fixes https://github.com/avo-hq/avo/issues/845

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/docs)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Manual review steps

Description...

1. Step 1
1. Step 2

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

File: spec/dummy/app/avo/resources/course_resource.rb [hunk 1]
```diff
@@ -8,6 +8,7 @@ class CourseResource < Avo::BaseResource
 
   field :id, as: :id
   field :name, as: :text
+  field :skills, as: :tags, blacklist: -> { record.skill_blacklist }, suggestions: -> { record.skill_suggestions }
```
Comment (rickychilcott) [inline spec/dummy/app/avo/resources/course_resource.rb:11]: Suggestion: Change this API for this field from `blacklist` to `disallowed`

Comment (adrianthedev) [reply spec/dummy/app/avo/resources/course_resource.rb:11]: Otherwise, how's is the field working? Does it fix your need?

**Comment by rickychilcott** [comment]
I'm not really having luck with this:


```ruby
  # Discussion Tags
  # model.discussion_tag_options =>
  # {"Test - Test"=>"gid://mission-met-center/Thredded::Messageboard/5",
  # "Chamber of Commerce - All Chambers of Commerce"=>"gid://mission-met-center/Thredded::Messageboard/3",
  # "Nonprofits - All Nonprofits"=>"gid://mission-met-center/Thredded::Messageboard/1",
  # "Cohort - THRIVE 2020.1"=>"gid://mission-met-center/Thredded::Messageboard/4",
  # "SBDC - All SBDCs"=>"gid://mission-met-center/Thredded::Messageboard/2",
  # "START - START"=>"gid://mission-met-center/Thredded::Messageboard/6"}
  field :discussion_tags,
    as: :select,
    options: ->(model:, resource:, view:, field:) { model.discussion_board_options }
  field :discussion_tags,
    as: :tags,
    suggestions: ->(model:, resource:, view:, field:) { model.discussion_board_options.keys }
```
only the `select` seems to render. Anything I'm missing with `tags`? I've tried various permiatations of `suggestions` and also just making it the example code (`field :skills, as: :tags, blacklist: -> { [] }, suggestions: -> { [] }`). Thoughts?

Screen Shot 2022-05-08 at 6 44 40 PM

**Comment by rickychilcott** [comment]
Yes, this is meeting my needs. Thank you! I sent a message not too long ago, but a restart of my webserver fixed the issue 🤪

The only thing I noticed is that if you provide suggestions in the array of hash form (i.e. `[{value: "abcd", label: "ABCD"}]`}, then the label should be used in the `show` action to resolve the labels for the selected options. Right now, it shows the values. Of course, some of the `values` might not be in the `suggestions`

**Comment by adrianthedev** [comment]
The `Show` view bug (showing the value, not the label) has been fixed in the released version. Also the blacklist -> disallowed suggestion was spot on.
Thank you!

**Comment by rickychilcott** [comment]
Thank you so much, Adrian! I've been out on vacation most of last week, so I got to play with the feature some and put into production, but not circle back to you.

This is great and thank you for all your work!

**Pull request merged**